src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -1427,18 +1427,24 @@ export interface TIX {
    * Place order for stocks.
    * @remarks
    * RAM cost: 2.5 GB
-   * Places an order on the stock market. This function only works for Limit and Stop Orders.
+   * Places a limit or stop order on the stock market. This function only works for Limit and Stop Orders.
    *
    * The ability to place limit and stop orders is **not** immediately available to the player and
    * must be unlocked later on in the game.
    *
    * Returns true if the order is successfully placed, and false otherwise.
    *
    * @param sym - Stock symbol.
-   * @param shares - Number of shares for order. Must be positive. Will be rounded to the nearest integer.
+   * @param shares - Number of shares for the order. Must be positive. Will be rounded to the nearest integer.
    * @param price - Execution price for the order.
-   * @param type - Type of order.
-   * @param pos - Specifies whether the order is a “Long” or “Short” position.
+   * @param type - Type of order. Valid values are:
+   *               - "Limit Buy"
+   *               - "Limit Sell"
+   *               - "Stop Buy"
+   *               - "Stop Sell"
+   * @param pos - Specifies whether the order is a “Long” or “Short” position. Valid values are:
+   *             - "Long"
+   *             - "Short"
```

Comment:
Unfortunately, the .md list syntax you're using gets messed up on the sub-pages that get generated by `npm run doc`. So we can't use this as-is. When I looked into it further, it turns out that the actual code that consumes the values is pretty lenient, so I suggest this:

```suggestion
   * @param type - Type of order. Must contain one of "limit" or "stop", and one of "buy" or "sell".
   * @param pos - Specifies whether the order is a “Long” or “Short” position. Must contain the letter "l" or "s".
```

Reply:
Can we use this PR as an opportunity to explicitly tell players the "expected" value? I really don't like the fuzzy matching of our old APIs. If we ever change the implementation (in a major version), it's best to tell them what we expect now.

Reply:
I suppose, except I'm not sure what exactly the expected value would be. The enum value is about the only thing that truly makes sense, but I don't think people will write out "Limit Buy Order".

Ultimately, I think we have to accept that the fuzzy matching on this one won't change.

Reply:
Using enum is a good idea.
```typescript
declare enum StockOrderType {
  LimitBuy = "Limit Buy",
  LimitSell = "Limit Sell",
  StopBuy = "Stop Buy",
  StopSell = "Stop Sell",
}

declare enum StockPosition {
  Long = "Long",
  Short = "Short",
}

placeOrder(
  sym: string,
  shares: number,
  price: number,
  type: StockOrderType | `${StockOrderType}`,
  pos: StockPosition | `${StockPosition}`,
): boolean;
```

I think this is fine. It does not break the backward compatibility, clearly tell players which value they should use, and the autocomplete feature can help them.

Reply:
We already have the enums PositionType and OrderType. I would use those rather than creating new, slightly different versions. 

I'm OK with lying slightly in the type in order to improve autocomplete and make a better dev experience, but I think I'm want to keep the comment the way I worded it in that case. It's very confusing when functions don't work the way they are specified; even if the way they work is not what we'd prefer, I think it's more important that the docs be correct.

Reply:
> We already have the enums PositionType and OrderType. I would use those rather than creating new, slightly different versions.

You are right. I did not see those two when I wrote the example.

> I'm OK with lying slightly in the type in order to improve autocomplete and make a better dev experience, but I think I'd want to keep the comment the way I worded it in that case. It's very confusing when functions don't work the way they are specified; even if the way they work is not what we'd prefer, I think it's more important that the docs be correct.

I understand.